Backpack/CardButton/Classes/CardButton.swift

```diff
@@ -0,0 +1,87 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ An Internal protocol that holds shared logic between ShareCardButton & SaveCardButton
+ */
+internal protocol CardButtonProtocol {
```

Comment:
If I created a parent class CardButton with shared logic, the parent class has to be public because the subclass BPKSaveCardButton will be public.

I want to prevent having a public CardButton class. This should be an internal class and not exposed to outside Backpack.

Is it okay to have a public class CardButton while we don't want it to be used (i.e. mark its constructor as private)?

Reply:
ahh, I understand!
Still it's only being used to configure the Background Circle view.
To me it seems we can extract the circular view as an internal component, like `CardButtonBackgroundView: UIView` and inside this class the size constraints would be sent and everything.

from the Save and Share classes you'd have a reference to this  `CardButtonBackgroundView` instead of just a UIView that you need to call 'configure' and 'setup' on.

Reply:
I created the CardButtonBackgroundView in 3cee10798cae8a1f998a84f849b8f8eef44746b9